src/network_modelling/network.py

```diff
@@ -196,21 +212,36 @@ def _remove_duplicates_buses_lines(self, buscoords):
         self.net.line.drop(duplicated_lines, inplace=True)
         self.net.bus.drop(duplicated_buses, inplace=True)
 
-    def pf_simulation(self, netp: list):
+    def pf_simulation(
+            self,
+            netp: list,
+            netp0: list = None,
+            netq_flex: list = None,
+            netq_non_flex: list = None):
         start = time.time()
         """ Given selected action, obtain voltage on buses and lines using pandapower """
-        self.net.load.p_mw = 0
-        self.net.sgen.p_mw = 0
-        for home in self.homes:
-            # Pandapower uses MW while the simulations uses kW
-            # Add a load if netp < 0 or a generation if netp > 0
-            if netp[home] >= 0:
-                self.net.load['p_mw'].iloc[home] = netp[home] / 1000
-            else:
-                self.net.sgen['p_mw'].iloc[home] = abs(netp[home]) / 1000
-
+        # removing old loads
+        for power in ['p_mw', 'q_var']:
+            self.net.load[power] = 0
+            self.net.sgen[power] = 0
+        # pandapower uses MW while the simulations uses kW
```

Comment:
should this comment be in `_assign_power_to_load_or_sgen`

Reply:
Yes, that's a good point. Thanks.